pkg/handlers/ocm.go

```diff
@@ -152,3 +157,86 @@ func BuildAndSendServiceLog(slBuilder *ServiceLogBuilder, firing bool, alert *te
 	}
 	return ocmClient.SendServiceLog(logEntry)
 }
+
+func (o *ocmClientImpl) SendLimitedSupport(clusterUUID string, lsReason *cmv1.LimitedSupportReason) error {
+	internalID, err := o.getInternalID(clusterUUID)
+	if err != nil {
+		return fmt.Errorf("can't get internal id: %w", err)
+	}
+
+	response, err := o.ocmConnection.ClustersMgmt().V1().Clusters().Cluster(internalID).LimitedSupportReasons().Add().Body(lsReason).Send()
+	if err != nil {
+		return fmt.Errorf("can't post limited support: %w", err)
+	}
+
+	// Check the response status code
+	if response.Status() < 200 && response.Status() >= 300 {
+		// Extract error details from the response and return an appropriate error.
+		return fmt.Errorf("unexpected status: %d", response.Status())
+	}
+
+	return nil
+}
+
+func (o *ocmClientImpl) RemoveLimitedSupport(clusterUUID string, lsReasonID string) error {
+	internalID, err := o.getInternalID(clusterUUID)
+	if err != nil {
+		return fmt.Errorf("can't get internal id: %w", err)
+	}
+
+	response, err := o.ocmConnection.ClustersMgmt().V1().Clusters().Cluster(internalID).LimitedSupportReasons().LimitedSupportReason(lsReasonID).Delete().Send()
+	if err != nil {
+		return fmt.Errorf("can't delete limited support reason %s from cluster %s: %w", lsReasonID, clusterUUID, err)
+	}
+
+	// Check the response status code
+	if response.Status() < 200 && response.Status() >= 300 {
+		// Extract error details from the response and return an appropriate error.
+		return fmt.Errorf("unexpected status: %d", response.Status())
+	}
+
+	return nil
+}
+
+func (o *ocmClientImpl) GetLimitedSupportReasons(clusterUUID string) ([]*cmv1.LimitedSupportReason, error) {
+
+	internalID, err := o.getInternalID(clusterUUID)
+	if err != nil {
+		return nil, fmt.Errorf("can't get internal id: %w", err)
+	}
+
+	response, err := o.ocmConnection.ClustersMgmt().V1().Clusters().Cluster(internalID).LimitedSupportReasons().List().Send()
+	if err != nil {
+		return nil, fmt.Errorf("can't get limited support reasons: %w", err)
+	}
+
+	// Check the response status code
+	if response.Status() < 200 && response.Status() >= 300 {
+		// Extract error details from the response and return an appropriate error.
+		return nil, fmt.Errorf("unexpected status: %d", response.Status())
+	}
+
+	return response.Items().Slice(), nil
+}
+
+func (o *ocmClientImpl) getInternalID(clusterUUID string) (string, error) {
```

Comment:
You can use the existing function from here: https://github.com/openshift/ocm-agent/blob/master/pkg/ocm/connection.go